ert3/storage/_storage.py

```diff
@@ -118,7 +118,54 @@ def _add_data(
         raise ert3.exceptions.StorageError(response.text)
 
 
-def _get_data(workspace: Path, experiment_name: str, record_name: str) -> Any:
+def _get_record_type(ensemble_record: ert3.data.EnsembleRecord) -> ert3.data.RecordType:
+    record_type = ensemble_record.records[0].record_type
+    for record in ensemble_record.records:
+        if record.record_type != record_type:
+            raise ValueError("Inconsistent record type")
+
+    return record_type
+
+
+def _add_numerical_data(
+    workspace: Path,
+    experiment_name: str,
+    record_name: str,
+    ensemble_record: ert3.data.EnsembleRecord,
+) -> None:
+    experiment = _get_experiment_by_name(experiment_name)
+    if experiment is None:
+        raise KeyError(
+            f"Cannot add {record_name} data to "
+            f"non-existing experiment: {experiment_name}"
+        )
+
+    metadata = {
+        "ensemble_size": ensemble_record.ensemble_size,
+        "record_type": _get_record_type(ensemble_record),
+    }
+    _add_file_data(
+        workspace, experiment_name, record_name + _METADATA, json.dumps(metadata)
+    )
+
+    ensemble_id = experiment["ensembles"][0]  # currently just one ens per exp
+    for idx, record in enumerate(ensemble_record.records):
+        df = pd.DataFrame([record.data], columns=record.index, index=[idx])
+        response = requests.post(
+            url=f"{_STORAGE_URL}/ensembles/{ensemble_id}/records/{record_name}/matrix",
+            params={"realization_index": idx},
+            data=df.to_csv().encode(),
+            headers={"content-type": "application/x-dataframe"},
```

Comment:
This's to be deprecated. use `text/csv` instead...

Reply:
It fails on `text/csv`. I know it is to be deprecated, but my impression is that as of now it does not work to use `text/csv`. Remember that this does not run on `ert-storage@master`...